### DISCUSSION TITLE
passing argv to vox

### DISCUSSION THREAD

**Discussion by ws1088** [xonsh/xonsh#4358] [Q&A] [answered]
Hi, I just want to write a xonsh script to take a variable number of arguments and pass it to vox. It seems like it is not so straightforward. The following does *NOT* work:

```py
#!/usr/bin/env xonsh
import sys
xontrib load vox
$a = sys.argv[1:]
vox $a

```

**Comment by jnoortheen** [comment]
I think the following would suffice your case

![image](https://user-images.githubusercontent.com/6702219/126319732-5a1ceaba-b141-4a22-a32c-e367d60fbe2a.png)

**Comment by anki-code** [comment, marked as answer]
You should use [Python evaluation](https://xon.sh/tutorial.html#python-evaluation-with):
```python
#!/usr/bin/env xonsh
import sys
xontrib load vox
vox @(sys.argv[1:])
```
Also you can use an [alias](https://xon.sh/tutorial.html#aliases) without wrapping into the script:
```python
xontrib load vox
aliases['supervox'] = "vox @($args)"
```
The simple example for testing with [`showcmd`](https://xon.sh/aliases.html#showcmd):
```python
aliases['ttt'] = "showcmd @($args)"
ttt 1 2 3
['1', '2', '3']
```
